### ISSUE TITLE
Add support for SITE_STREET and SITE_LOCALITY matchPrecisions

### ISSUE THREAD

**Issue by mraross** [bcgov/ols-geocoder#128]
The matchPrecisionValues.csv file contains values for two proposed matchPrecisions: SITE_STREET and SITE_LOCALITY. This is to distinguish and reward site matches that are associated with addresses at different precision levels as follows:

matchPrecision|value|Example
SITE|100|Robson Arms -- 1324 Robson St, Vancouver, BC
SITE_STREET|97|Centennial Candle -- Laurel Lane, Victoria, BC
SITE_LOCALITY|94|Mill Bay (locality) -- Gingolx, BC

SITE is considered the highest (civic number) level of precision.

**Comment by mraross** [comment]
Another use of the new matchPrecisions is keeping sites in order in autocomplete results when the user doesn't enter a frontGate. For example, say the user enters: 

  Delta

If the geocoder is enhanced to treat frontGate as optional, it will stick a frontGate on the end of addressString and match the following:

  North Delta -- Delta, BC
  Delta Port Exit -- Roberts Bank Rd, Delta, BC
  Delta Creek Bridge -- Terrace, BC
  Delta Port Entrance -- Roberts Bank Rd, Delta, BC
  Delta, BC

While the matches are good, their order isn't. siteStreet matches should come before siteLocality matches as in:

  Delta Port Exit -- Roberts Bank Rd, Delta, BC
  Delta Port Entrance -- Roberts Bank Rd, Delta, BC
  Delta Creek Bridge -- Terrace, BC
  North Delta -- Delta, BC
  Delta, BC

New matchPrecision hierarchy
province | locality | siteLocality | street | siteStreet | block | civicNumber | site, unit

Does siteLocality = subLocality ?

Why is there a Mill Bay -- Ferry Rd, Mill Bay, BC ?

**Comment by mraross** [comment]
Does siteLocality = subLocality ?

Why is there a Mill Bay -- Ferry Rd, Mill Bay, BC ?

**Comment by gleeming** [comment]
It is a GSR NCAP record for the Mill Bay ferry terminal.

**Comment by mraross** [comment]
As discussed, could you please cut this occupant and non-civic address out of the August data you are using so that it doesn't cause confusion when we try to show ICBC that we have disambiguated the bianyms. I will contact GeoBC and get them to change the name from Mill Bay to Mill Bay Ferry Terminal.

**Comment by mraross** [comment]
Tsawwassen, BC shows up twice; once as a landing (at the ferry terminal), and once as a community. Also there is one civic address:

     1 Ferry Causeway, Delta, BC

which implies that Tsawassen (landing) should be a site within Delta and not a DRA locality?

**Comment by gleeming** [comment]
I have removed the ferry terminal record for Mill Bay from my GSR source file.

In terms of Tsawwassen, I'm not clear on the implication or what you are asking.  There are two records from BCGNIS and these are disambiguated by including the feature types: Tsawwassen (community) -- Delta, BC and Tsawwassen (landing) -- Delta, BC.  They are non-civic site records within Delta and therefore associated with ITN's Delta locality.   1 Ferry Causeway, Delta, BC is a BC Assessment civic address also associated with ITN's Delta locality.  There is no Tsawwassen locality in ITN, only Delta and Washington in this general area.

**Comment by mraross** [comment]
You are correct. Tsawwassen is handled perfectly in 4.1. I was looking at 4.0. Sorry for the confusion. As for the Mill Bay issue, please review issue

**Comment by mraross** [comment]
If I enter the following into Bronze:

     20 IR 9 -- Dease Lake, BC

I do get the correct result although it is ranked second after:

     Dease Lake 9 -- Dease Lake, BC

The order can only be controlled if we add a new matchPrecision of SITE_LOCALITY with a value less than that of BLOCK and greater than 90. If we add in SITE_BLOCK as well, SITE_LOCALITY should be lower than SITE_BLOCK. For example:

|Match Precision|Value|
|---|---|
SITE|100
UNIT|100
CIVIC_NUMBER|100
BLOCK|99
SITE_BLOCK|98
SITE_LOCALITY|97

**Comment by gleeming** [comment]
The FME processing scripts will need to be updated, in particular the occupant geocoding interpretation.

**Comment by mraross** [comment]
I have reviewed the concept of site and matchPrecision in the context of supporting sublocalities within a locality.
|Site type| Identifier|Example|Associated matchPrecision|
|---|---|---|---|
|civic site|
|non-civic sites|
|subLocalities|

A civic site is identified by a civic number on a given street. An address that matches to a civic site is assigned a matchPrecision of CIVIC_NUMBER

A non-civic site is identified by a site name on a given street. An address that matches down to the non-civic site is assigned a matchPrecision of SITE.
A sublocality is a site within a locality and is identified by a site name that is within or near a given DRA locality and the name of the DRA locality

**Comment by mraross** [comment]
Under my latest proposal, when you enter a subLocality without the locality as in the following:

     Mill Bay

The geocoder should return the following matches:

|Matched Address|Score|matchPrecision|Faults|
|---|---|----|---|
Mill Bay|67|LOCALITY(68)|Missing provinceCode (-1)
Mill Bay|63|SUB_LOCALITY(74)|Missing locality (-10), missing provinceCode (-1)

**Comment by cmhodgson** [comment]
> We are planning to treat all BCGNIS localityNames that aren't DRA localities as sites within a DRA locality. 

Is this still true, is this what we want to do?

Besides the answer to that, I don't think adding in a new matchPrecision with a different bas score is the right approach, I actually think we should do away with matchPrecision base scores because they are hindering other kinds of matches (eg. intentional locality matches like "Bear Bay 8, BC").
